Scripts/Decorator/WaitForCondition.cs

```diff
@@ -28,6 +28,23 @@ public WaitForCondition(Func<bool> condition, Node decoratee) : base("WaitForCon
         }
 
         protected override void DoStart()
+        {
+            addTimerOrStartImmediately();
+        }
+
+        public override void Pause()
+        {
+            base.Pause();
+            Clock.RemoveTimer(checkCondition);
+        }
+
+        public override void Resume()
+        {
+            base.Resume();
+            addTimerOrStartImmediately();
```

Comment:
I think this might cause it to start the child node even though the `base.Resume` node might already have started it. You will need to do some more checking here. 
You only want to call the `addTimerOrStartImmediately` if the Decoratee object wasn't yet started.